soroban-env-host/src/events/internal.rs

```diff
@@ -75,34 +75,41 @@ pub enum InternalDiagnosticArg {
     XdrVal(ScVal),
 }
 
-fn externalize_args(host: &Host, args: &[InternalDiagnosticArg]) -> Result<Vec<ScVal>, HostError> {
-    let mut scargs: Vec<ScVal> = Vec::new();
-    for arg in args.iter() {
-        match arg {
-            InternalDiagnosticArg::HostVal(h) => scargs.push(host.from_host_val(*h)?),
-            InternalDiagnosticArg::XdrVal(v) => scargs.push(v.clone()),
-        }
-    }
-    Ok(scargs)
-}
-
 impl InternalDiagnosticEvent {
-    pub fn to_xdr(&self, host: &Host) -> Result<xdr::ContractEvent, HostError> {
-        let topics: xdr::VecM<ScVal> = externalize_args(host, &self.topics)?.try_into()?;
-        let args = externalize_args(host, &self.args)?;
-        let data = if args.len() > 1 {
-            ScVal::Vec(Some(xdr::ScVec::from(xdr::VecM::try_from(args)?)))
-        } else if let Some(arg) = args.into_iter().next() {
-            arg
-        } else {
-            ScVal::Void
-        };
-        Ok(xdr::ContractEvent {
-            ext: xdr::ExtensionPoint::V0,
-            contract_id: self.contract_id.clone(),
-            type_: xdr::ContractEventType::Diagnostic,
-            body: xdr::ContractEventBody::V0(xdr::ContractEventV0 { topics, data }),
-        })
+    fn to_xdr(&self, host: &Host) -> Result<Option<xdr::ContractEvent>, HostError> {
+        // this exist as an internal closure to prevent it from being accidentally
+        // called from the outside without `with_debug_budget`
```

Comment:
why would it matter if this was called outside `with_debug_budget`?

Reply:
I mainly wanted to make sure whatever logic that exist as a function that meant to be in debug mode is wrapped around `with_debug_budget`. 
I moved this closure out and added the `is_debug` check on entry.